src/blockdata/script.rs

```diff
@@ -216,6 +216,57 @@ impl Script {
     /// Creates a new empty script
     pub fn new() -> Script { Script(vec![].into_boxed_slice()) }
 
+    /// Generates P2PK-type of scriptPubkey
+    pub fn with_pk(pubkey: &PublicKey) -> Script {
```

Comment:
This should be named `p2pk` or something similar to indicate that you are making a pay-to-pubkey scriptpubkey. The `with_` naming is confusing here.

Reply:
Right. I've taken `with_` prefix after Rust naming guidelines https://rust-lang.github.io/api-guidelines/naming.html suggesting to name constructors as `new` or `with_` prefix. But I agree that this is confusing and guidelines are not a law.

Renamed all methods to corresponding descriptor names with `new_` prefix to signify that we are not converting existing script object into a scriptPubkey but constructing a new one from arguments.